src/Database/PostgreSQL/PQTypes/Internal/C/Types.hsc

```diff
@@ -236,6 +239,62 @@ instance Storable PGbytea where
 
 ----------------------------------------
 
+-- PGuuid C type is *char, but in actual expects the array
+-- to be 16 bytes. The bytes are encoded directly from the
+-- hex representation of the UUID, which is different from
+-- the default binary encoding from Data.UUID.
+data PGuuid = PGuuid {
```

Comment:
`data UUID` from https://hackage.haskell.org/package/uuid-types-1.0.3/docs/Data-UUID-Types.html is the same as this one and it also provides Storable instance that is said to be compatible with RFC (which seems to be different than yours btw). Is there a reason for not using it here?

Reply:
I tested with a few UUID encoding libraries in Haskell. IIRC the problem is libpostgres does _not_ follow the standard RFC encoding so none of them work out. I copied the encoding algorithm from some other postgres library and it was the only one that worked out. My memory is a bit hazy and I'd probably have to dig out what approaches I've tried.

Reply:
I probably copied it from here:
  - https://github.com/nikita-volkov/postgresql-binary/blob/master/library/PostgreSQL/Binary/Decoding.hs#L157
  - https://github.com/nikita-volkov/postgresql-binary/blob/master/library/PostgreSQL/Binary/Encoding/Builders.hs#L162